library/core/src/main/java/com/google/android/exoplayer2/offline/DownloadHelper.java

```diff
@@ -723,6 +727,61 @@ public DownloadRequest getDownloadRequest(String id, @Nullable byte[] data) {
     return new DownloadRequest(id, downloadType, uri, streamKeys, cacheKey, data);
   }
 
+  /**
+   * Try to estimate the total size of chunked downloads. Returns {@link C#LENGTH_UNSET} for
```

Comment:
There is probably no reason to special-case progressive streams because you already allow `C.LENGTH_UNSET` as return value if "not known". Progressive streams can just go through the same code paths allowing for the possibility that a bitrate is set in the Format so that we can actually calculate the total size.

Reply:
The reason for this special case is that progressive streams don't call `prepare()`, and as a result, `assertPreparedWithMedia()` fails. I figured that in this case `LENGTH_UNSET` is better than a crash. But that depends on ExoPlayer's policy regarding this type of violation.

Reply:
Thanks for explaining. Can you use "mediaSource == null" as an alternative check? That's what we use in other places.